src/sage/geometry/cone_critical_angles.py

```diff
@@ -0,0 +1,978 @@
+
+    Note that we don't return a basis for each eigenspace along with
+    its eigenvalue. For the application we have in mind, an eigenspace
+    of dimension greater than one (so, ``multiplicity > 1``) is an
+    error. As such, our return value is optimized for convenience in
+    the non-error case, where there is only one eigenvector (spanning
+    a one-dimensional eigenspace) associated with each eigenvalue.
+
+    ALGORITHM:
+
+    According to Proposition 5 [Or2020]_, the solutions corresponding
+    to non-zero eigenvalues can be found by solving a smaller
+    eigenvalue problem in only the variable `\xi`. So, we do that, and
+    then solve for `\eta` in terms of `\xi` as described in the
+    proposition.
+
+    EXAMPLES:
+
+    When the zero solutions are included, this function returns the
+    same solutions as the naive method on the Schur cone in three
+    dimensions::
+
+        sage: from itertools import chain
+        sage: from sage.geometry.cone_critical_angles import (
+        ....:   _normalize_gevp_solution,
+        ....:   _solve_gevp_naive,
+        ....:   gevp_licis,
+        ....:   solve_gevp_nonzero,
+        ....:   solve_gevp_zero)
+        sage: K = cones.schur(3)
+        sage: gs = [g.change_ring(AA).normalized() for g in K]
+        sage: G = matrix.column(gs)
+        sage: GG = G.transpose()*G
+        sage: G_index_sets = list(gevp_licis(G))
+        sage: all( set( _normalize_gevp_solution(s) for s in
+        ....:              chain(solve_gevp_zero(GG,I,J),
+        ....:                    solve_gevp_nonzero(GG,GG,GG,I,J)) )
+        ....:      ==
+        ....:      set( _normalize_gevp_solution(s) for s in
+        ....:            _solve_gevp_naive(GG,GG,GG,I,J) )
+        ....:      for I in G_index_sets
+        ....:      for J in G_index_sets )
+        True
+
+    TESTS:
+
+    This function should return the same solutions (with zero included,
+    of course) as the naive implementation even for random cones::
+
+        sage: # long time
+        sage: from itertools import chain
+        sage: from sage.geometry.cone_critical_angles import (
+        ....:   _normalize_gevp_solution,
+        ....:    _random_admissible_cone,
+        ....:   _solve_gevp_naive,
+        ....:   gevp_licis,
+        ....:   solve_gevp_nonzero,
+        ....:   solve_gevp_zero)
+        sage: n = ZZ.random_element(1,3)
+        sage: P = _random_admissible_cone(ambient_dim=n)
+        sage: Q = _random_admissible_cone(ambient_dim=n)
+        sage: gs = [g.change_ring(AA).normalized() for g in P]
+        sage: G = matrix.column(gs)
+        sage: GG = G.transpose()*G
+        sage: hs = [h.change_ring(AA).normalized() for h in Q]
+        sage: H = matrix.column(hs)
+        sage: HH = H.transpose()*H
+        sage: M = G.transpose()*H
+        sage: G_index_sets = list(gevp_licis(G))
+        sage: H_index_sets = list(gevp_licis(H))
+        sage: all( set( _normalize_gevp_solution(s) for s in
```

Comment:
Beware the style police

Reply:
I'm not sure what you meant so I've reindented these two examples and have probably made them worse.

Reply:
I may have accidentally made this acceptable to PEP8. In any case, the style police haven't filed any charges.